mesonbuild/backend/ninjabackend.py

```diff
@@ -1057,9 +1057,10 @@ def generate_target(self, target):
         self.add_build(elem)
         #In AIX, we archive shared libraries. If the instance is a shared library, we add a command to archive the shared library
         #object and create the build element.
-        if isinstance(target, build.SharedLibrary) and self.environment.machines[target.for_machine].is_aix():
-            elem = NinjaBuildElement(self.all_outputs, linker.get_archive_name(outname), 'AIX_LINKER', [outname])
-            self.add_build(elem)
+        if isinstance(target, build.SharedLibrary) and not isinstance(target, build.SharedModule) and self.environment.machines[target.for_machine].is_aix():
+            if target.aix_so_archive is True:
```

Comment:
What if we set `aix_so_archive = False` as a property on SharedModule, to override the one on SharedLibrary?

Reply:
Thank you @eli-schwartz for your review. 

This is actually better. So if we set aix_so_archive variable to False in SharedModule and override it, we will not have to do any checks in ninjabackend.py for SharedModule at all. It will just override it and when py3.extension_module is called, aix_so_archive will be False there, as a consequence we will not need to archive. 

Amazing!! Thanks once again @eli-schwartz for your guidance. When I checked this with glib2, scipy and p11 kit, this logic worked. I will soon commit again with these changes. 
 
 Have a nice day ahead.